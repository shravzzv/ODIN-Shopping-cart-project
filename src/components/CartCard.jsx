import '../styles/components/CartCard.css'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

CartCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
}

export default function CartCard({ id, image, title, quantity, price }) {
  const handleRemove = () => {}

  return (
    <div className='card elevated cartCard'>
      <Link to={`/product/${id}`}>
        <img src={image} alt={title} />
      </Link>
      <div className='content'>
        <h1 className='title'>{title}</h1>
        <p className='price'>${price}</p>
        <button className='filled-tonal' onClick={handleRemove}>
          Remove
        </button>
        <div className='badgeContainer'>
          <span className='badgeLabel'>
            {quantity >= 1000 ? '999+' : quantity}
          </span>
        </div>
      </div>
    </div>
  )
}
