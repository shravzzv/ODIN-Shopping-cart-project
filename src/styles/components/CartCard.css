.cartCard {
  display: flex;
  align-items: center;
  gap: 5px;
  cursor: auto;

  a {
    flex: 1;
    display: flex;

    img {
      border-radius: 12px;
    }
  }

  .content {
    padding: 5px;
    flex: 1;
    align-self: stretch;
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .title {
      color: var(--primary);
      font-family: inherit;
      font-size: 1.125rem;
      margin: 0;
      text-align: center;
    }

    .badgeContainer {
      position: absolute;
      top: 0;
      right: 0;
      background-color: var(--error);
      border-radius: 9999px;
      padding: 4px;

      span {
        color: var(--on-error);
        font-size: 16px;
        font-weight: 600;
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }
  }
}

@media (max-width: 767px) {
}

@media (min-width: 768px) {
}

@media (min-width: 1024px) {
}
