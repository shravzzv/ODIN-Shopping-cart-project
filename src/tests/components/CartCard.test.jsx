import { describe, test, expect } from 'vitest'
import { screen, render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import CartCard from '../../components/CartCard'

describe('Cart card', () => {
  test('should display an image, a title, a price, a quantity and a remove button', () => {
    render(
      <MemoryRouter>
        <CartCard
          id={1}
          image='/products/suit1a.webp'
          title='Midnight Black Tuxedo'
          quantity={2}
          price={599}
        />
      </MemoryRouter>
    )
    expect(screen.getByRole('img')).toBeInTheDocument()
    expect(screen.getByRole('img').alt).toBe('Midnight Black Tuxedo')

    expect(screen.getByText('Midnight Black Tuxedo')).toBeInTheDocument()
    expect(screen.getByText('Midnight Black Tuxedo')).toHaveClass('title')

    expect(screen.getByText('$599')).toBeInTheDocument()
    expect(screen.getByText('$599')).toHaveClass('price')

    expect(screen.getByText('2')).toBeInTheDocument()
    expect(screen.getByText('2')).toHaveClass('badgeLabel')

    expect(screen.getByRole('button', { name: /remove/i })).toBeInTheDocument()
  })

  test('should display the quantity as 999+ if the quantity is > 999', () => {
    const { container } = render(
      <MemoryRouter>
        <CartCard
          id={1}
          image='/products/suit1a.webp'
          title='Midnight Black Tuxedo'
          quantity={2000}
        />
      </MemoryRouter>
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByText('999+')).toBeInTheDocument()
    expect(screen.queryByText('2000')).not.toBeInTheDocument()
  })
})
