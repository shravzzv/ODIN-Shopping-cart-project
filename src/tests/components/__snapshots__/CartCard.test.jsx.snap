// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart card > should display the quantity as 999+ if the quantity is > 999 1`] = `
<div>
  <div
    class="card elevated cartCard"
  >
    <a
      href="/product/1"
    >
      <img
        alt="Midnight Black Tuxedo"
        src="/products/suit1a.webp"
      />
    </a>
    <div
      class="content"
    >
      <h1
        class="title"
      >
        Midnight Black Tuxedo
      </h1>
      <p
        class="price"
      >
        $
      </p>
      <button
        class="filled-tonal"
      >
        Remove
      </button>
      <div
        class="badgeContainer"
      >
        <span
          class="badgeLabel"
        >
          999+
        </span>
      </div>
    </div>
  </div>
</div>
`;
